app/src/main/java/chat/rocket/android/service/RocketChatWebSocketThread.java

```
@@ -238,6 +252,35 @@ public void run() {
                   return null;
                 });
 
+                task.getResult().client.getOnFailureCallback().onSuccess(_task -> {
+                  ddpClient = null;
+                  CompositeDisposable subscriptions = new CompositeDisposable();
+                  connectivityManager.notifyConnecting(hostname);
+                  subscriptions.add(
+                          RxJavaInterop.toV2Single(connect().retry())
+                                  .subscribe(
+                                          rocketChatWebSocketThread -> {
+                                            String roomId = rocketChatCache.getSelectedRoomId();
+                                            if (roomId != null) {
+                                              StreamRoomMessage streamRoomObserver = new StreamRoomMessage(
+                                                      appContext,
+                                                      hostname,
+                                                      realmHelper,
+                                                      ddpClientRef,
+                                                      roomId
+                                              );
+                                              streamRoomObserver.register();
+                                              listeners.add(streamRoomObserver);
+                                            }
+                                          },
+                                          err -> {
+                                            subscriptions.dispose();
```

Don't ignore the exception. Is it safe to assume that no more subscriptions will be added to the CompositeDisposable after this?